### ISSUE TITLE
[serve] test_backend_state.py fails locally and not tested in bazel

### ISSUE THREAD

**Issue by simon-mo** [ray-project/ray#14395]
I can't run https://github.com/ray-project/ray/blob/master/python/ray/serve/tests/test_backend_worker.py locally because it faied with 
```mock_backend_state_inputs = (<ray.serve.backend_state.BackendState object at 0x7f877838c080>, <MagicMock name='RayInternalKVStore' id='140219814406968'>, <MagicMock name='LongPollHost' id='140219814406128'>, <MagicMock name='AsyncGoalManager' id='140219814284984'>)

    def test_completed_goals_delta_backend(mock_backend_state_inputs):
        backend_state = mock_backend_state_inputs[0]
        b1 = "backend_one"
>       backend_state.backends[b1] = generate_mock_backend_info()
E       AttributeError: 'BackendState' object has no attribute 'backends'
```

and it turns out it's not included in the BUILD file so it's not tested in our CI:
https://github.com/ray-project/ray/blob/master/python/ray/serve/BUILD

**Comment by simon-mo** [comment]
test_async_goal_manager.py is not included in build as well and it cannot be ran in py3.6
```
pytest -v -s test_async_goal_manager.py 
Test session starts (platform: darwin, Python 3.6.9, pytest 5.3.5, pytest-sugar 0.9.2)
cachedir: .pytest_cache
benchmark: 3.2.2 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /Users/simonmo/Desktop/ray/ray/python
plugins: sugar-0.9.2, rerunfailures-9.1.1, asyncio-0.10.0, custom-exit-code-0.3.0, benchmark-3.2.2, aiohttp-0.3.0, timeout-1.4.2, cov-2.8.1
collecting ... 

―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― test_wait_for_goals ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

    @pytest.mark.asyncio
    async def test_wait_for_goals():
        manager = AsyncGoalManager()
    
        # Check empty goal
        await manager.wait_for_goal(None)
    
        goal_id = manager.create_goal()
>       waiting = asyncio.create_task(manager.wait_for_goal(goal_id))
E       AttributeError: module 'asyncio' has no attribute 'create_task'

test_async_goal_manager.py:15: AttributeError

 ray/serve/tests/test_async_goal_manager.py::test_wait_for_goals ⨯                                                                 100% ██████████

Results (0.12s):
       1 failed
         - ray/serve/tests/test_async_goal_manager.py:7 test_wait_for_goals
```

**Comment by edoakes** [comment]
@simon-mo thanks for pointing these out. I'll fix `test_async_goal_manager`

**Comment by simon-mo** [comment]
fixed in #14419
